cli/src/config-schema.json

```diff
@@ -112,6 +112,14 @@
                         "tool": {
                             "type": "string",
                             "description": "External tool for generating diffs"
+                        },
+                        "tool-mode": {
+                            "description": "Invoke the external tool with directories or individual files",
+                            "enum": [
+                                "dir",
+                                "file-by-file"
+                            ],
+                            "default": "dir"
```

Comment:
I think we'll also need per-tool default/supported modes or per-tool/mode args.
https://github.com/martinvonz/jj/issues/3696#issuecomment-2116426056

Reply:
Good idea, switch to per-tool only.